src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/TokenBasedFormattingRule.cs

```diff
@@ -508,8 +508,13 @@ private static SyntaxList<UsingDirectiveSyntax> GetUsings(SyntaxNode node)
             }
 
             // pointer case for regular pointers
-            if ((currentToken.Kind() == SyntaxKind.AsteriskToken && currentToken.Parent is PointerTypeSyntax) ||
-                (previousToken.Kind() == SyntaxKind.AsteriskToken && previousToken.Parent is PrefixUnaryExpressionSyntax))
+            if (currentToken.Kind() == SyntaxKind.AsteriskToken && currentToken.Parent is PointerTypeSyntax)
+            {
+                return CreateAdjustSpacesOperation(1, AdjustSpacesOption.ForceSpacesIfOnSingleLine);
```

Comment:
❗ I believe this will break the formatter for types like `int**` (it will start to format them as `int* *`.

Reply:
@sharwell Do we want to fix both? or only `NormalizeWhitespace`?

Reply:
shouldn't we always adjust this to 0 spaces if we have the `*` of a pointer type? i would think we would always want zero spaces between that and whatever comes before it

Reply:
@CyrusNajmabadi Whops, I mixed things up. Swapped the expected behavior and the actual behavior.
The formatter behaves as expected. This is only a NormalizeWhitespace issue.
Closing the PR.